### PR TITLE
scryer-prolog: 0.9.0 -> 0.9.1

### PR DESCRIPTION
**Pull request by wkral**
###### Description of changes
Update to new release of scryer-prolog.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wkral** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scryer-prolog</li>
  </ul>
</details>

**Comment by triska** [comment]
Why does `openssl` still appear in `Cargo.toml` in the upgraded version? It does not appear in the `Cargo.toml` that ships with Scryer&nbsp;Prolog. One of the features of the new release is that it eliminates the dependency on OpenSSL.

File: pkgs/development/compilers/scryer-prolog/cargo.patch [hunk 1]
```diff
@@ -1,66 +1,24 @@
...
-index 8e81a947..d21a8d49 100644
+index 6e42de23..a50e3d47 100644
 --- a/Cargo.toml
 +++ b/Cargo.toml
-@@ -55,7 +55,7 @@ ring = "0.16.13"
- ripemd160 = "0.8.0"
- sha3 = "0.8.2"
- blake2 = "0.8.1"
--openssl = { version = "0.10.29", features = ["vendored"] }
-+openssl = { version = "0.10.29" }
```
Comment (triska) [inline pkgs/development/compilers/scryer-prolog/cargo.patch:46]: This line is unexpected: It does not appear in `Cargo.toml` of Scryer Prolog.

Comment (wkral) [reply pkgs/development/compilers/scryer-prolog/cargo.patch:46]: This is a diff of a patch file so it adds a layer of complexity, this view of the patch is what was being used with 0.9.0. But the changes to OpenSSL are being removed here as it's no longer in the Cargo.toml. The patch is still needed to use the system built libraries for gmp, mpc, and mpfr.

Comment (triska) [reply pkgs/development/compilers/scryer-prolog/cargo.patch:46]: Perfect, thank you a lot!

File: pkgs/development/compilers/scryer-prolog/default.nix [hunk 2]
```diff
@@ -13,30 +13,21 @@
...
-      url = "https://patch-diff.githubusercontent.com/raw/mthom/scryer-prolog/pull/1342.patch";
-      sha256 = "2N0AOkFuf+H/aUn2QTXgmqjmvShTxHxB6kNuNdNoVRI=";
-    })
   ];
 
-  cargoSha256 = "nqAHVXAmTW9mdE2L2yhpOTz16JbYgQUmCgiFq9pBzUU=";
+  cargoSha256 = "A6HtvxGTjJliDMUSGkQKB13FRyfBU4EPvrlZ97ic0Ic=";
 
   nativeBuildInputs = [ pkg-config rustfmt];
   buildInputs = [ openssl gmp libmpc mpfr ];
```
Comment (malbarbo) [inline pkgs/development/compilers/scryer-prolog/default.nix:33]: openssl must be remove here.

File: pkgs/development/compilers/scryer-prolog/cargo.patch (hunk 1 above)
Comment (malbarbo) [inline pkgs/development/compilers/scryer-prolog/cargo.patch:46]: The patch must be regenerated to remove this line.

**Review by malbarbo — CHANGES_REQUESTED** [review]
Please, remove the openssl dependency.

File: pkgs/development/compilers/scryer-prolog/default.nix (hunk 2 above)
Comment (wkral) [reply pkgs/development/compilers/scryer-prolog/default.nix:33]: There is still an indirect dependency on openssl through the native-tls lib.

File: pkgs/development/compilers/scryer-prolog/cargo.patch (hunk 1 above)
Comment (wkral) [reply pkgs/development/compilers/scryer-prolog/cargo.patch:46]: The patch was regenerated, and this is the diff between the last patch and the current. All of these lines here are being removed from the patch, so the additon and the removal of openssl are both being removed from the patch, and there is no reference to openssl in the Cargo.toml resulting patch file with the diff applied.

File: pkgs/development/compilers/scryer-prolog/default.nix (hunk 2 above)
Comment (malbarbo) [reply pkgs/development/compilers/scryer-prolog/default.nix:33]: You are right, sorry. I saw [this](https://github.com/mthom/scryer-prolog/commit/9c1de8e00bcf8cf110066fc2a5aff93625fdf9d1) commit and thought that the openssl dependency was completely removed.

File: pkgs/development/compilers/scryer-prolog/cargo.patch (hunk 1 above)
Comment (malbarbo) [reply pkgs/development/compilers/scryer-prolog/cargo.patch:46]: You are right again.

**Review by malbarbo — APPROVED** [review]
LGTM, thanks @wkral.

File: pkgs/development/compilers/scryer-prolog/default.nix [hunk 3]
```diff
@@ -13,30 +13,21 @@
...
   };
 
   cargoPatches = [
     # Use system openssl, gmp, mpc and mpfr.
     ./cargo.patch
-
-    ./fix-tests.patch
-
-    # Avoid testing failing with "couldn't save history"
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scryer-prolog/default.nix:32]: Please remove the unused fetchpatch from inputs.

Comment (wkral) [reply pkgs/development/compilers/scryer-prolog/default.nix:32]: Done, thanks for catching that.

**Pull request merged**